gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java

```diff
@@ -90,6 +90,8 @@ public class FileAwareInputStreamDataWriter extends InstrumentedDataWriter<FileA
   public static final boolean DEFAULT_GOBBLIN_COPY_CHECK_FILESIZE = false;
   public static final String GOBBLIN_COPY_TASK_OVERWRITE_ON_COMMIT = "gobblin.copy.task.overwrite.on.commit";
   public static final boolean DEFAULT_GOBBLIN_COPY_TASK_OVERWRITE_ON_COMMIT = false;
+  public static final String GOBBLIN_COPY_REQUIRE_PERMISSION_SET_FOR_SUCCESS = "gobblin.copy.requirePermissionSetForSuccess";
+  public static final boolean DEFAULT_COPY_REQUIRE_PERMISSION_SET_FOR_SUCCESS = false;
```

Comment:
when wouldn't one want failure?  I'd be more aggressive in choosing the default.  while it does require a config change, one can always easily revert to the legacy semantics.  hence not b/w-compat, but easily opted-into, as desired

Reply:
Sure, I was also debating making the default true instead. I'll change.